test/listen.3.test.js

```diff
@@ -15,73 +15,69 @@ before(async function () {
 
 // https://nodejs.org/api/net.html#net_ipc_support
 if (os.platform() !== 'win32') {
-  test('listen on socket', t => {
-    t.plan(3)
+  test('listen on socket', async t => {
+    t.plan(2)
     const fastify = Fastify()
-    t.teardown(fastify.close.bind(fastify))
+    t.after(() => fastify.close())
 
     const sockFile = path.join(os.tmpdir(), `${(Math.random().toString(16) + '0000000').slice(2, 10)}-server.sock`)
     try {
       fs.unlinkSync(sockFile)
     } catch (e) { }
 
-    fastify.listen({ path: sockFile }, (err, address) => {
-      t.error(err)
-      t.strictSame(fastify.addresses(), [sockFile])
-      t.equal(address, sockFile)
-    })
+    await fastify.listen({ path: sockFile })
+    t.assert.deepStrictEqual(fastify.addresses(), [sockFile])
+    t.assert.strictEqual(fastify.server.address(), sockFile)
   })
 } else {
-  test('listen on socket', t => {
-    t.plan(3)
+  test('listen on socket', async t => {
+    t.plan(2)
     const fastify = Fastify()
-    t.teardown(fastify.close.bind(fastify))
+    t.after(() => fastify.close())
 
     const sockFile = `\\\\.\\pipe\\${(Math.random().toString(16) + '0000000').slice(2, 10)}-server-sock`
 
-    fastify.listen({ path: sockFile }, (err, address) => {
-      t.error(err)
-      t.strictSame(fastify.addresses(), [sockFile])
-      t.equal(address, sockFile)
-    })
+    await fastify.listen({ path: sockFile })
+    t.assert.deepStrictEqual(fastify.addresses(), [sockFile])
+    t.assert.strictEqual(fastify.server.address(), sockFile)
   })
 }
 
-test('listen without callback with (address)', t => {
+test('listen without callback with (address)', async t => {
   t.plan(1)
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
-  fastify.listen({ port: 0 })
+  t.after(() => fastify.close())
+  await fastify.listen({ port: 0 })
     .then(address => {
-      t.equal(address, `http://${localhostForURL}:${fastify.server.address().port}`)
+      t.assert.strictEqual(address, `http://${localhostForURL}:${fastify.server.address().port}`)
     })
 })
 
-test('double listen without callback rejects', t => {
+test('double listen without callback rejects', async t => {
   t.plan(1)
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
-  fastify.listen({ port: 0 })
+  t.after(() => fastify.close())
+  await fastify.listen({ port: 0 })
```

Comment:
This could become a flaky test:

- the inner `fastify.listen({ port: 0 })` does not return the promise to the parent promise chain (`await fastify.listen({ port: 0 })`) so we may hit a (rare) case where the first `listen` resolve the promise and the test fails because the `t.assert.ok(err)` is not yet executed.

To simplify it we should use the `done` callback instead

Reply:
Done!, see commit here:[140700f](https://github.com/fastify/fastify/pull/6022/commits/140700f3a02e2848775880b5d8410b786f1686da)